src/core/storage/index.js

```diff
@@ -0,0 +1,46 @@
+// @flow
+// $FlowIgnore[missing-export]
+import {TextEncoder, TextDecoder} from "util";
+
+/**
+ * A byte string represented as a Uint8Array. Each item in a byteString array
+ * is a Uint8 representing a byte value.
+ */
+export type ByteString = Uint8Array;
+
+/**
+ * Data Storage is meant as a low-level interface to data that is not
+ * already in process memory. If the value for the key entered cannot be found,
+ * an error should throw.
```

Comment:
Can we at least mention something about keys being filesystem-friendly?
We can flesh out the details later.